File: WebViewDemo/WebViewDemo/ViewController.m
//
//  ViewController.m
//  WebViewDemo
//
//  Created by wanshaofa on 15/1/29.
//  Copyright (c) 2015年 enuke. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()<UIWebViewDelegate>
@property (weak, nonatomic) IBOutlet UIWebView *webView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    NSString *filePath = [[NSBundle mainBundle] pathForResource:@"Untitled.html" ofType:nil];
    NSString *htmlStr = [[NSString alloc] initWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:nil];
    [self.webView loadHTMLString:htmlStr baseURL:nil];
}


- (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)request navigationType:(UIWebViewNavigationType)navigationType{
    NSURL *url = [request URL];
    
    NSLog(@"%@", [url absoluteString]);
    return YES;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: InfixOperateDemo/InfixOperateDemo/ViewController.swift
//
//  ViewController.swift
//  InfixOperateDemo
//
//  Created by wanshaofa on 15/2/4.
//  Copyright (c) 2015年 enuke. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let firstVector = 2
        let secondVector = 3
        let plusMinusVector = firstVector +*- secondVector
        
        NSLog("%d", plusMinusVector)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
        
    }
    
    
//    infix operator + { associativity left precedence 140 }
//    func + (left:Int,right:Int){
//    
//    }

}


infix operator +*- { associativity left precedence 140 }
func +*- (left: Int, right: Int) -> Int {
    return (left + right ) * (left - right)
}